File: project/assets/scripts.js
// Copyright 2024 Joshua Rich <joshua.rich@gmail.com>.
// SPDX-License-Identifier: 	AGPL-3.0-or-later

// htmx
import 'htmx.org'
import './setup-htmx.js'

// hyperscript
import _hyperscript from 'hyperscript.org'
_hyperscript.browserInit()

import { themeChange } from 'theme-change'
themeChange()

File: web/content/scripts.hs
-- Copyright 2026 Joshua Rich <joshua.rich@gmail.com>.
-- SPDX-License-Identifier: 	AGPL-3.0-or-later

-- GetQueryParam will fetch the value of the given query parameter. If no query parameter exists with the given name, an
-- empty string is returned.
def GetQueryParam(name)
  make a URLSearchParams from the search of the document's location called query
  if query.has(name)
    set value to the query.get(name)
    return value
  else
    return ''
  end
end
